src/lib/matchers.js

```diff
@@ -85,7 +85,17 @@ const getQueryStringMatcher = ({ query: expectedQuery }) => {
 		const query = querystring.parse(getQuery(url));
 		debug('  Expected query parameters:', expectedQuery);
 		debug('  Actual query parameters:', query);
-		return keys.every((key) => query[key] === expectedQuery[key]);
+		return keys.every((key) => {
+			const value = query[key];
+			const expectedValue = expectedQuery[key];
+			if (Array.isArray(value) && Array.isArray(expectedValue)) {
```

Comment:
What about if my mock is `key: ['val']`. I _think_ this won't match `?key=val` as the parsed query string will be `{key: 'val'}`

So needs a test for that scenario. And I think the clearest way to write the matcher is to coerce both value and expectedValue to arrays up front, then compare the arrays

Reply:
True, I think it should be okay now. Looks a bit cleaner too.